lib/chef/knife/vsphere_vm_snapshot.rb

```diff
@@ -71,7 +102,11 @@ def run
     end
 
     if config[:create_new_snapshot]
-      vm.CreateSnapshot_Task(:name => config[:create_new_snapshot], :description => "", :memory => false, :quiesce => false)
+      if get_config(:memory)
+      	vm.CreateSnapshot_Task(:name => config[:create_new_snapshot], :description => "", :memory => true, :quiesce => false)
+      else
+      	vm.CreateSnapshot_Task(:name => config[:create_new_snapshot], :description => "", :memory => false, :quiesce => false)
```

Comment:
I like this addition, and I could see adding a similar option for `quiesce`. I'm not 100% sure if you need to set a default on the option, but you could just pass `get_config(:memory)` directly instead of the `if`, as it should be either `true` or `false`:

```
vm.CreateSnapshot_Task(name: config[:create_new_snapshot], description: "", memory: get_config(:memory), quiesce: false)
```